project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -864,6 +939,105 @@ record ExperimentUpdateResponse(String experimentId, ExperimentUpdateResponseBod
 
   }
 
+  /**
+   * A service request to create a new experimental group.
+   *
+   * @param projectId    the project's identifier. The project containing the experiment.
+   * @param experimentId the experiment's identifier
+   * @param group        the experimental group to create
+   * @param requestId    the request ID. Needs to be provided by the client and will be referenced
+   *                     in the response.
+   * @since 1.10.0
+   */
+  record ExperimentalGroupCreationRequest(String projectId, String experimentId,
+                                          ExperimentalGroup group, String requestId) implements
+      CacheableRequest {
+
+    public ExperimentalGroupCreationRequest(String projectId, String experimentId,
+        ExperimentalGroup group) {
+      this(projectId, experimentId, group, UUID.randomUUID().toString());
+    }
+  }
+
+  /**
+   * A service response from a {@link ExperimentalGroupCreationRequest}.
+   *
+   * @param experimentId the experiment's identifier
+   * @param group        the experimental group created
+   * @param requestId    the identifier of the original request to which this is a response.
+   * @since 1.10.0
+   */
+  record ExperimentalGroupCreationResponse(String experimentId, ExperimentalGroup group,
+                                           String requestId) {
+
+  }
+
+  /**
+   * A service request to update an experimental group.
+   *
+   * @param projectId    the project's identifier. The project containing the experiment.
+   * @param experimentId the experiment's identifier
+   * @param group        the experimental group to update
+   * @param requestId    the request ID. Needs to be provided by the client and will be referenced
+   *                     in the response.
+   * @since 1.10.0
+   */
+  record ExperimentalGroupUpdateRequest(String projectId, String experimentId,
```

Comment:
You probably want to implement the compact constructor as well to check for null request ids. Records come with the all args constructor by default.